### ISSUE TITLE
Overwatch doest not refresh at 144hz on lutris wine 6.10-2

### ISSUE THREAD

**Issue by Ondra3211** [lutris/wine#85]
Hey, I updated wine to to lutris-6.10-2 and I noticed that game is no refreshing at 144hz even if it is set to 144hz.
Changing to bordless mode and going back to fullscreen mode didnt't help.
This happens only with the new lutris-6.10.-2 version

**Comment by tannisroot** [comment]
How did you determine that it's not refreshing at that refreshrate?

**Comment by Ondra3211** [comment]
Simply by my eyes. The game is just not smooth. There is no difference if I set 60 hz or 144 hz. In game I have around 200 fps so it should be smooth with 144 hz but it's not. This is only on lutris-6.10-2.

**Comment by SpookySkeletons** [comment]
Can confirm, the frame delivery is not smooth in 6.10-2.
It's so terrible it feels more like 30-40 fps gameplay.

**Comment by Ondra3211** [comment]
Still same with Lutris Wine 6.10-3 and Lutris Wine 6.10-3-fshack

**Comment by tannisroot** [comment]
I can't reproduce this issue, I have my monitor overclocked to 75hz and there is a clear difference when i set it to that refreshrate.
What Desktop Environment do you use and what backend (I assume X11?).
Also, does it happen with older versions?

**Comment by Ondra3211** [comment]
Operating System: Manjaro Linux
KDE Plasma Version: 5.22.1
KDE Frameworks Version: 5.83.0
Qt Version: 5.15.2
Kernel Version: 5.12.10-1-MANJARO (64-bit)
Graphics Platform: X11
Processors: 12 × AMD Ryzen 5 3600 6-Core Processor
Memory: 15,6 GiB of RAM
Graphics Processor: Radeon RX 570 Series

Also, does it happen with older versions?
No, as I said this only with the new lutris6.10-x version. Going back to lutris 6-4 or 5.7 is completely fine.

**Comment by tannisroot** [comment]
Does this happen with regular Wine-staging 6.10?

**Comment by tannisroot** [comment]
> Can confirm, the frame delivery is not smooth in 6.10-2.
> It's so terrible it feels more like 30-40 fps gameplay.
> 
> Want to add this is an Nvidia card. The GPU usage seems to drop, some kind of CPU bottleneck causing choppy frame delivery. Almost certainly seems to be the lutris build.

Show me a screenshot of the game with DXVK_HUD envar set here with these values:
![image](https://user-images.githubusercontent.com/10602045/122944453-8379bc00-d380-11eb-890a-5c350e513c3f.png)

**Comment by Ondra3211** [comment]
> > ```
> > Operating System: Manjaro Linux
> > KDE Plasma Version: 5.22.1
> > KDE Frameworks Version: 5.83.0
> > Qt Version: 5.15.2
> > Kernel Version: 5.12.10-1-MANJARO (64-bit)
> > Graphics Platform: X11
> > Processors: 12 × AMD Ryzen 5 3600 6-Core Processor
> > Memory: 15,6 GiB of RAM
> > Graphics Processor: Radeon RX 570 Series
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also, does it happen with older versions?
> > No it doesnt. As I said this only with the new lutris6.10-x version. Going back to lutris 6-4 or 5.7 is completely fine.
> 
> Also, does it happen with lutris-6.1-3?

lutris-6.1-3? No
Wine-staging 6.10? yes

**Comment by Ondra3211** [comment]
I moved to Kubuntu to test if the problem still remain and it is. I tried almost all wine version but still same. Maybe it is a problem in KDE. Probably it started by upgrading to Plasma 5.22. @tannisroot do you use gnome or plasma? I can try to downgrade to Plasma 5.21 and try if it works.

Should changing refresh rate in Overwatch work? If I change refresh rate in Overwatch nothing happens.
In lutriy 5.7, 6.0 the game feels like 144hz but still I feel some lags.
Any other wine version feels like 60hz.

**Comment by tannisroot** [comment]
> I tried almost all wine version but still same
What wine versions and from which sources?

Could you switch back to Manjaro and install `downgrade` utility from AUR and determine, based on your existing findings (6.1 is good, while 6.10 is bad), the first wine-staging version in which this issue appeared, and the last version in which it worked normally?

**Comment by Ondra3211** [comment]
> > I tried almost all wine version but still same
> 
> What wine versions and from which sources?
> 
> Could you switch back to Manjaro and install `downgrade` utility from AUR and determine, based on your existing findings (6.1 is good, while 6.10 is bad), the first `wine-staging` version in which this issue appeared, and the last version in which it worked normally?

lutris 6.1-3 is smooth on my 144hz but wine staging 6.1 is laggy as any other version of wine I tried.
Only lutris 5.7 and 6.1 is working for me.

**Comment by Ondra3211** [comment]
With wine staging 6.12 I am not even able to launch the game. Fixes mentioned [here](https://github.com/lutris/docs/blob/master/Battle.Net.md#sleeping-agent--issues-with-installing--updating-) doesn't work. 
![Screenshot_20210708_003245](https://user-images.githubusercontent.com/39314277/124837464-289db280-df85-11eb-96a9-2905499de377.png)

**Comment by Ondra3211** [comment]
Running game with manually suspended compositor and in borderless mode than the game is running at 144hz. But as of perfomance it's not the best as lutris 6.1. Like if I move a lot or running in game than it's still not smooth as lutris 6.1 or lutris 5.7.

Lutris 6.1 and 5.7 suspend the compositor automatically and it is smooth even in fullscreen mode. That's not happening with other wine versions.

**Comment by tannisroot** [comment]
> Running game with manually suspended compositor and in borderless mode than the game is running at 144hz.
Can you rephrase that?

**Comment by Ondra3211** [comment]
> > Running game with manually suspended compositor and in borderless mode than the game is running at 144hz.
> > Can you rephrase that?
I disabled KDE compositor by pressing ALT+SHIFT+F12
Then I launched the Game, went to settings and changed display mode from fullscreen to borderless windowed.

Lutris 6.1 disabling compositor automaticlly and it doesnt matter if display mode is set to fullscreen or borderless mode.
That's really nice because I don't need to reneable the composing each time I alt-tab from the game.

**Comment by Ondra3211** [comment]
So the problem is that wine doesnt suspend the compositor as lutris 6.1. I have to do it manually.

**Comment by Ondra3211** [comment]
On Wayland the 6.10 is running totally fine.
On Xorg I have to disable composing (alt+shift+f12)
